Issue 163 path

VladimirAlexiev opened the pull request:
Closes #163

* [See this document rendered online here](https://raw.githack.com/VladimirAlexiev/data-shapes/issue-163-PATH/shacl12-sparql/index.html)
* @HolgerKnublauch Sorry, I made a github mistake and this includes a gazillion commits. Can you pick only the last one https://github.com/w3c/data-shapes/pull/620/commits/8d3d82338b48fef9563fc14c46756b7f01d9fd1a ?
* I've deleted a few duplicated words from a definition and replaced with a reference to the section that defines "substitution" fully.
* Added suggestion to emit `"BAD-PATH"`
* Added some words in Security as per @afs

VladimirAlexiev committed:
minor editorial change

VladimirAlexiev committed:
plantuml version of diagram

VladimirAlexiev committed:
plantuml version of diagram

VladimirAlexiev committed:
plantuml version of diagram

VladimirAlexiev committed:
add manual `<color>` and `<u>` tags to class names

VladimirAlexiev committed:
add manual `<color>` and `<u>` tags to class names

VladimirAlexiev committed:
add manual `<color>` and `<u>` tags to class names

VladimirAlexiev committed:
add manual `<color>` and `<u>` tags to class names

VladimirAlexiev committed:
Update shacl12-core/images/SHACL-UML.puml

Co-authored-by: Yousouf Taghzouti <48333491+YoucTagh@users.noreply.github.com>

VladimirAlexiev committed:
Update shacl12-core/images/SHACL-UML.puml

Co-authored-by: Yousouf Taghzouti <48333491+YoucTagh@users.noreply.github.com>

VladimirAlexiev committed:
Update shacl12-core/images/SHACL-UML.puml

Co-authored-by: Yousouf Taghzouti <48333491+YoucTagh@users.noreply.github.com>

VladimirAlexiev committed:
Update shacl12-core/images/SHACL-UML.puml

Co-authored-by: Yousouf Taghzouti <48333491+YoucTagh@users.noreply.github.com>

VladimirAlexiev committed:
- sh:order: xsd:integer or xsd:decimal
- sh:path : predicate or list or blank node and link to #property-paths
- add ellipsis as per TallTed
- rdf:List -> "list" because SHACL doesn't require the rdf:List type
- add sh:nodeKind, sh:ignoredProperties

VladimirAlexiev committed:
- sh:order: xsd:integer or xsd:decimal
- sh:path : predicate or list or blank node and link to #property-paths
- add ellipsis as per TallTed
- rdf:List -> "list" because SHACL doesn't require the rdf:List type
- add sh:nodeKind, sh:ignoredProperties

VladimirAlexiev committed:
add <color:blue><u> to PropertyGroup

VladimirAlexiev committed:
add <color:blue><u> to PropertyGroup

VladimirAlexiev committed:
replace textual diagram with PlantUML SVG diagram

VladimirAlexiev committed:
replace textual diagram with PlantUML SVG diagram

VladimirAlexiev committed:
replace <img> with script that embeds SVG content

VladimirAlexiev committed:
replace <img> with script that embeds SVG content

VladimirAlexiev committed:
replace <img> with script that embeds SVG content

VladimirAlexiev committed:
Merge branch 'w3c:gh-pages' into gh-pages

VladimirAlexiev committed:
Merge branch 'w3c:gh-pages' into gh-pages

VladimirAlexiev committed:
Merge branch 'w3c:gh-pages' into gh-pages

bergos committed:
added sh:conformanceDisallows, updated conformance-definition

bergos committed:
improved conformance definition language, fixed tabs vs. spaces

bergos committed:
added not about def conformance disallow set, updated severity section

bergos committed:
Apply suggestions from code review

Co-authored-by: Alex Nelson <alexander.nelson@nist.gov>

bergos committed:
improved naming for conformance-disallow set, describe default set

bergos committed:
Update shacl12-core/index.html

Co-authored-by: Alex Nelson <alexander.nelson@nist.gov>

bergos committed:
added conformanceDisallows to changes section

bergos committed:
copied datashapes.org widgets to ui spec

bergos committed:
fixed widget anchor links

ajnelson-nist committed:
Add tests demonstrating conformanceDisallows

Test contents adapted from Core document example by @bergos.

Test structure adapted from this file, picked more or less at random:
shacl12-test-suite/tests/core/path/path-alternative-001.ttl

Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

bergos committed:
updated dash namespaces to shui

bergos committed:
removed TQ specific text, small cleanup

YoucTagh committed:
Update shacl12-core/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

YoucTagh committed:
Update shacl12-core/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

YoucTagh committed:
Update shacl12-core/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

YoucTagh committed:
add a note for test suite conformance

bergos committed:
Apply suggestions from code review

Co-authored-by: Ieben Smessaert <smessie@users.noreply.github.com>
Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>
Co-authored-by: Edmond Chuc <37032744+edmondchuc@users.noreply.github.com>

nicholascar committed:
minor updates to show use of definitions

nicholascar committed:
tidy editors, add alex

nicholascar committed:
HTML reformat; packaging section addition; terminology listing updates

bergos committed:
change ui widget score to ul lists

nicholascar committed:
conformance section

nicholascar committed:
ontology containers

nicholascar committed:
typo

nicholascar committed:
typo

bergos committed:
improved wording

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
file + ontology IRI note

nicholascar committed:
remove static defined term listing - use index auto list

nicholascar committed:
move Spec Profiles to Appendix

VladimirAlexiev committed:
add core-components-list (3 fields), sh:targetWhere, sh:codeIdentifier. Move sh:path to top of PropertyShape

VladimirAlexiev committed:
add core-components-list (3 fields), sh:targetWhere, sh:codeIdentifier. Move sh:path to top of PropertyShape

VladimirAlexiev committed:
restore link to PropertyGroup

VladimirAlexiev committed:
restore link to PropertyGroup

VladimirAlexiev committed:
Merge branch 'w3c:gh-pages' into gh-pages

VladimirAlexiev committed:
fixes #163

VladimirAlexiev committed:
fixes #163

afs commented:
>  a github mistake and this includes a gazillion commits.

PRs are merged, not single commits from a PR.

You can get the commit with, for example, `git show 8d3d823 > F`, then create a clean, up-to-date branch, `git apply F` and  submit a PR with the required changes.

HolgerKnublauch reviewed (changes requested):
Yeah I am afraid the best way forward would be for you to either create a new PR with just the change that you want, or roll back the unneeded parts here.

nicholascar committed:
add in ref to US issues

nicholascar committed:
inc UCs in rec section

nicholascar committed:
up to dependency calcs

nicholascar committed:
link calc deps to motivation

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Ieben Smessaert <smessie@users.noreply.github.com>

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Ieben Smessaert <smessie@users.noreply.github.com>

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Ieben Smessaert <smessie@users.noreply.github.com>

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Ieben Smessaert <smessie@users.noreply.github.com>

bergos committed:
made letter after rendering capital in ui spec

HolgerKnublauch committed:
#614: Attempt to work around outdated W3C validator complaints (#621)

nicholascar committed:
move to local copies of CSS files (#628)

HolgerKnublauch committed:
Added style.css to core too

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
apply several of Ted's changes

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
further application of Ted's changes

smessie committed:
Issue 534: Support for property paths in SHACL UI (#600)

* Define support for property paths in SHACL UI

* Remove targets from the examples

Apply suggestions from code review

Co-authored-by: Thomas Bergwinkl <bergos@users.noreply.github.com>

* Add notes on restrictions and challenges

* Add shape example to "Other Complex Paths" subsection

* Don't put RFC2119 keywords in bold

* Replace repetitive RFC2119 keywords

---------

Co-authored-by: Thomas Bergwinkl <bergos@users.noreply.github.com>

HolgerKnublauch committed:
#629 (#630)

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Ieben Smessaert <smessie@users.noreply.github.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

HolgerKnublauch committed:
#209: Added clarifying paragraph that node shapes should use rdfs:label/comment (#632)

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Matt Goldberg <59745812+mgberg@users.noreply.github.com>

nicholascar committed:
update to match Matt's previous comment

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Matt Goldberg <59745812+mgberg@users.noreply.github.com>

nicholascar committed:
tidy shape example id

nicholascar committed:
Fix ReSpec display errors

MPvHarmelen committed:
Add 'sh:resultMessage' to some tests (#642)

HolgerKnublauch committed:
GH-587: shnex:nodesWhere  node expression (#611)

* #587: Added shnex:where

* Formatting (including unrelated examples)

* Renamed from shnex:where to shnex:nodesWhere

* Applied PR feedback

* Renamed nodesWhere to nodesMatching

* Update index.html

Changed sh:targetMatching to sh:targetWhere.

---------

Co-authored-by: Robert David <9427084@gmail.com>

edmondchuc committed:
Add property roles section

HolgerKnublauch committed:
#653: Fixed broken link

HolgerKnublauch committed:
Replaced duplicate conform with conforms

bergos committed:
#645: [UI] added terminology

simonstey committed:
#455: Integrate SPARQL Functions (#511)

* feat: :sparkles: automatically fetch sparql functions and create sections dynamically

* style: :art: restore formatting of sections

* Import SPARQL function triples into shnex-sparql.ttl — labels, descriptions, categories, spec links.

* Render SPARQL functions table in node-expr UI from local shnex-sparql.ttl

* feat: Add examples for SPARQL functions in shnex-sparql vocabulary

- Introduced examples for various SPARQL functions to enhance documentation and usability.
- Added a new prefix for shnex-sparql to categorize examples.
- Updated existing functions with example usage, including arithmetic, logical, string, numeric, and date functions.
- Ensured all aggregate functions also include example usage for clarity.

* chore: Normalize SPARQL function labels to lowercase in shnex-sparql.ttl; fix isdefinedby URIs

* refactor: :speech_balloon: Replace dc:description with rdfs:comment

* feat: add property definitions for function categorization, example and infix operator symbols. Add infix operators to selected functions

* fix: update comments for clarity in SPARQL functions and update .gitignore entries

* fix: :twisted_rightwards_arrows: prepare rebase with HEAD

* fix: clean up HTML structure; fix merge issues

* Minor cleanup to align indentation, and make sure error message doesn't wipe out complete section

* Merged Description and Example into one column to (hopefully) improve layout

* Corrections to some examples

* Update shacl12-vocabularies/shnex-sparql.ttl

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Update shacl12-vocabularies/shnex-sparql.ttl

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Replaced rdf:types of functions to follow SHACL conventions

The classes sparql:Function etc are ill-defined and serve IMHO no purpose from a SHACL point of view. By replacing them with sh:ListParameterExpressionFunction we follow SHACL syntax and also make the owl:import to the external namespace unnecessary. (That owl:import does not resolve right now anyway)

* Deleted aggregate functions because they belong into shnex as they take lists as parameters

* Removed filter-exists, not-exists, in and not-in

These do not fit the execution model of node expressions. If people still want to use them, they can use SPARQL strings via sh:select etc.

* Tabs vs spaces

* Updated JS to new rdf:type

* Minor fixes

---------

Co-authored-by: Holger Knublauch <holger@topquadrant.com>
Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>
Co-authored-by: Robert David <9427084@gmail.com>

edmondchuc committed:
Apply suggestion from @bergos and @TallTed

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

edmondchuc committed:
Apply suggestion from @bergos and @TallTed

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

edmondchuc committed:
Apply @TallTed suggestions from code review

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

dependabot[bot] committed:
Bump js-yaml from 4.1.0 to 4.1.1 in /scripts

Bumps [js-yaml](https://github.com/nodeca/js-yaml) from 4.1.0 to 4.1.1.
- [Changelog](https://github.com/nodeca/js-yaml/blob/master/CHANGELOG.md)
- [Commits](https://github.com/nodeca/js-yaml/compare/4.1.0...4.1.1)

---
updated-dependencies:
- dependency-name: js-yaml
  dependency-version: 4.1.1
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

HolgerKnublauch committed:
#647: Added at risk issue to pre-binding (#656)

* #647: Added at risk issue to pre-binding

* Update shacl12-sparql/index.html

Co-authored-by: Andy Seaborne <andy@apache.org>

* Update shacl12-sparql/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

---------

Co-authored-by: Andy Seaborne <andy@apache.org>
Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

edmondchuc committed:
Apply suggestions from code review

edmondchuc committed:
add note about RDF model compatibility

bergos committed:
fix: #646 removed severity count, aligned violation naming

bergos committed:
chore: #663 moved built-in widget to separate section

edmondchuc committed:
Update shacl12-ui/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

edmondchuc committed:
Update shacl12-ui/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

HolgerKnublauch committed:
#648: Moved table of severities below syntax rules, added fallback for sh:message in SPARQL (#667)

* #648: Moved table of severities below syntax rules, added fallback for sh:message in SPARQL

* Fixed typo

* Update shacl12-core/index.html

Co-authored-by: Andy Seaborne <andy@apache.org>

---------

Co-authored-by: Andy Seaborne <andy@apache.org>

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Edmond Chuc <37032744+edmondchuc@users.noreply.github.com>

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Edmond Chuc <37032744+edmondchuc@users.noreply.github.com>

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Edmond Chuc <37032744+edmondchuc@users.noreply.github.com>

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

afs committed:
Definitions for rule set evaluation

afs committed:
Negation syntax: NOT

danielbeeke committed:
Add new team member Daniel Beeke to index.html

afs committed:
Fix for description of SHACL-SPARQL

afs committed:
Replace partial use of tabs by spaces

afs committed:
Update shacl12-rules/shacl-rules.ttl

Co-authored-by: simon <simon.steyskal@siemens.com>

afs committed:
Updates from review of shacl12-rules/index.html

Co-authored-by: simon <simon.steyskal@siemens.com>

afs committed:
shnex: namespace

HolgerKnublauch committed:
#496: Replaced shnex and shui namespaces (#681)

afs committed:
Placeholders for SHACL Rules tests

nicholascar committed:
improve the SHACL-SHACL links - using shacl-shacl namespace URI

YoucTagh committed:
#594: moved the note about the test suite to a shared document

nicholascar committed:
give profile definitions

nicholascar committed:
creating other profiles

caribouW3 committed:
Update index.html

caribouW3 committed:
Update specifications.html

link fix for compact syntax

nicholascar committed:
start of profile hierarchies

nicholascar committed:
profiling standards & data placeholders

bergos committed:
Update shacl12-ui/index.html

Co-authored-by: Edmond Chuc <37032744+edmondchuc@users.noreply.github.com>

afs committed:
Initial SRL tests

YoucTagh committed:
#594: use the link of the new test suite

HolgerKnublauch committed:
#649: Fixed definitions to singular, added uniqueness (#668)

* #649: Fixed definitions to singular, added uniqueness

* Update shacl12-sparql/index.html

Co-authored-by: Andy Seaborne <andy@apache.org>

* Deleted unused definitions of Binding and Solution from CORE

Also some minor formatting clean up and removal of an outdated TODO

---------

Co-authored-by: Andy Seaborne <andy@apache.org>

afs committed:
Correct links

HolgerKnublauch committed:
#650: Added link to VarExpression to clarify what "variable" means here (#655)

* #650: Added link to VarExpression to clarify what "variable" means here

* Removed link again, and the redundant mention of focusNode, also fixed a duplicate HTML id

* Switched from using 'node' to 'term', fixed indentation

simonstey committed:
refactor: rename union/minus node expressions to join/remove in node … (#636)

* refactor: rename union/minus node expressions to join/remove in node expression spec

* feat: add advanced sequence operations (flatMap, findFirst, matchAll) to node expressions

* docs: add newlines and minor whitespace improvements in node expressions spec

* docs: update SHACL vocabulary comments to reflect join/remove terminology

* refactor: rename minus/union node expressions to remove/join in shnex vocabulary

* feat: add vocabulary definitions for flatMap, findFirst, and matchAll node expressions

* docs: clarify distinction between sh:path and shnex:pathValues in node expressions spec

* docs: remove redundant description from Advanced Sequence Operations section

* chore: apply fixes proposed by Ted

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* chore: apply suggested fix to wording

* fix: replace leftover 'union' with 'join'

* refactor: rename Join expressions to Concat expressions in vocabulary and documentation

* enhance: add examples for shnex:concat and shnex:remove expressions in documentation

* enhance: add evaluation traces for various sequence operations

* fix: update deprecation comments for sh:union to reference shnex:concat

* Apply suggestions from code review

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>
Co-authored-by: Holger Knublauch <holger@topquadrant.com>

---------

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>
Co-authored-by: Holger Knublauch <holger@topquadrant.com>

HolgerKnublauch committed:
#649: Added link to #defn_sparqlSolutionMapping (#695)

Also replaced direct hyperlinks with data-cite and removed unnecessary / before #

HolgerKnublauch committed:
Several formatting fixes and improvements (#696)

* Several formatting fixes and improvements

* Fixed grammar, slightly different variation from David's suggestion

afs committed:
Set spec locations to www.w3.org/TR/

nicholascar committed:
fix KurrawongAI links

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Scott Henninger <scotthenninger@gmail.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-profiling/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

HolgerKnublauch committed:
#703: Fixed plural in rdfs:comment of sh:NodeShape (#705)

* #703: Fixed plural in rdfs:comment of sh:NodeShape

* On this occasion also addressing https://github.com/w3c/data-shapes/pull/668#discussion_r2620487984

This was a comment on an already closed PR.

caribouW3 committed:
Update index.html (#714)

Editors draft link fix

caribouW3 committed:
Update index.html

caribouW3 committed:
Update index.html

caribouW3 committed:
use respec dfn syntax for links to other specs (#715)

Relative links to point to other specs does not work in a simple <a> element, Respec has the dfn element to refer to concepts from other specs.

caribouW3 committed:
Update index.html

typo

HolgerKnublauch committed:
#702: Changed definition of sh:deactivated to mean "ignored" or skipped. (#708)

HolgerKnublauch committed:
#712: Switched to sh:targetClass, added note on how to handle this test (#713)

* #712: Switched to sh:targetClass, added note on how to handle this test

Also removed the sh:resultMessage as that's implementation-specific.

* Also sneaking in some follow-up errors from the batch namespace replacement

* And one more fix, mostly due to using the old example.org namespace

HolgerKnublauch committed:
Issue 706: Rewritten Abstract (#716)

* #706: Possible starting point for new abstract

* Iterated on abstract based on Andy's input

* Applied a few comments and switched from software systems to "agents" that may include human agents

* Update shacl12-core/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Corrected merge

* Update shacl12-core/index.html

Co-authored-by: Andy Seaborne <andy@apache.org>

* Simplified abstract

* Applied Livio's feedback on abstract

* Update shacl12-core/index.html

Co-authored-by: Andy Seaborne <andy@apache.org>

---------

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>
Co-authored-by: Andy Seaborne <andy@apache.org>

HolgerKnublauch committed:
#709: Added sh:unit property (#717)

* #709: Added sh:unit property

* Update shacl12-core/index.html

Co-authored-by: Andy Seaborne <andy@apache.org>

* Update shacl12-vocabularies/shacl.ttl

Co-authored-by: Yousouf Taghzouti <48333491+YoucTagh@users.noreply.github.com>

* Generalized sh:unit to node shapes

* Added explicit reference to QUDT

* Generalized sh:unit to allow IRIs

* Apply suggestions from code review

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Applied almost all PR feedback

---------

Co-authored-by: Andy Seaborne <andy@apache.org>
Co-authored-by: Yousouf Taghzouti <48333491+YoucTagh@users.noreply.github.com>
Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

afs committed:
Update rule grammar

afs committed:
Remove Rule3 rule form (SRL)

HolgerKnublauch committed:
#491: Added node expression test cases and clarified spec in two places (#719)

* #491: Added node expression test cases and clarified spec in two places

* Added readme to explain the format of tests

* Trying to revert line break changes

* Added example to readme

* Apply suggestions from code review

Co-authored-by: Alex Nelson <alexander.nelson@nist.gov>

* Applied more of Alex's feedback

* Addressed various TODOs and added proper example for shnex:sum (also as a test)

* Added test and definition of shnex:orderBy

Also restored alphabetic order in shnex, causing a larger diff (but only the stuff about shnex:orderBy is new)

* Apply suggestions from code review

Co-authored-by: Robert David <9427084@gmail.com>

---------

Co-authored-by: Alex Nelson <alexander.nelson@nist.gov>
Co-authored-by: Robert David <9427084@gmail.com>

dependabot[bot] committed:
Bump undici and @rdfjs/serializer-jsonld-ext in /scripts

Bumps [undici](https://github.com/nodejs/undici) and [@rdfjs/serializer-jsonld-ext](https://github.com/rdfjs-base/serializer-jsonld-ext). These dependencies needed to be updated together.

Updates `undici` from 5.29.0 to 6.23.0
- [Release notes](https://github.com/nodejs/undici/releases)
- [Commits](https://github.com/nodejs/undici/compare/v5.29.0...v6.23.0)

Updates `@rdfjs/serializer-jsonld-ext` from 4.0.1 to 4.0.2
- [Commits](https://github.com/rdfjs-base/serializer-jsonld-ext/compare/v4.0.1...v4.0.2)

---
updated-dependencies:
- dependency-name: undici
  dependency-version: 6.23.0
  dependency-type: indirect
- dependency-name: "@rdfjs/serializer-jsonld-ext"
  dependency-version: 4.0.2
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

afs committed:
More SHACL Rules syntax tests

afs committed:
Wellformedness, stratification, eval tests

afs committed:
GH-728: Inference graph

caribouW3 committed:
Create auto-publish-shacl12-rules.yml

Auto-publication configuration

HolgerKnublauch committed:
Issue 723 724 (#738)

* #723 #724

Also allowing rdf:dirLangString where rdf:langString was allowed.

Also allowing rdf:HTML at sh:message and sh:description.

* Added commas

dependabot[bot] committed:
Bump lodash from 4.17.21 to 4.17.23 in /scripts

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.21 to 4.17.23.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.21...4.17.23)

---
updated-dependencies:
- dependency-name: lodash
  dependency-version: 4.17.23
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

afs committed:
Auto-publish for rules and node-expr

caribouW3 committed:
Revert "Create auto-publish-shacl12-rules.yml"

afs committed:
RDF domain and range eval tests

HolgerKnublauch committed:
* #676: Rewrote security section and used MUST for now

* Update shacl12-sparql/index.html

Co-authored-by: Andy Seaborne <andy@apache.org>

* Switched to informative "should", accepted most of Ted's prose changes.

* Update shacl12-sparql/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Update shacl12-sparql/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

---------

Co-authored-by: Andy Seaborne <andy@apache.org>
Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

afs committed:
GH-743: Remove some SRL functions

afs committed:
Define negation element evaluation

simonstey committed:
Apply suggestion from @simonstey

afs commented:
Ping @VladimirAlexiev

HolgerKnublauch committed:
#276: Fixed prefix declarations in test cases and adjusted examples to use .com instead of .org (#605)

HolgerKnublauch committed:
Issue 737 (#750)

* #737

* Added change log entry

* Update shacl12-core/index.html

Co-authored-by: Andy Seaborne <andy@apache.org>

* Update shacl12-core/index.html

Co-authored-by: Andy Seaborne <andy@apache.org>

* Update shacl12-core/index.html

Co-authored-by: Andy Seaborne <andy@apache.org>

* Apply suggestion from @afs

Co-authored-by: Andy Seaborne <andy@apache.org>

* Update shacl12-core/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Update shacl12-core/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Update shacl12-core/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

---------

Co-authored-by: Andy Seaborne <andy@apache.org>
Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

HolgerKnublauch committed:
#657: Clarified that members of list must be literals/IRIs and the same order is kept (#710)

* #657: Clarified that members of list must be literals/IRIs and the same order is kept

* Update shacl12-node-expr/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Update shacl12-node-expr/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Switched from ex:concat to sparql:concat

* Apply suggestions from code review

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Update shacl12-node-expr/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Update shacl12-node-expr/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

* Update shacl12-node-expr/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

---------

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

YoucTagh committed:
Revise SHACL 1.2 Profiling abstract

robert-david committed:
Resolved rebase conflicts

afs commented:
"This branch is [30 commits ahead of](https://github.com/VladimirAlexiev/data-shapes/compare/w3c%3Adata-shapes%3Agh-pages...issue-163-PATH) and [187 commits behind](https://github.com/VladimirAlexiev/data-shapes/compare/issue-163-PATH...w3c%3Adata-shapes%3Agh-pages)"

Would it be possible to rebase this PR with the current gh-pages?

afs commented on shacl12-core/index.html:
```diff
@@ -932,7 +932,23 @@ <h2>Shapes and Constraints</h2>
+				The content of the boxes under the class name lists some of the properties that instances of these classes may have, together with their value types.
+				Empty arrowheads indicate inheritance (<code>rdfs:subClassOf</code>) and normal arrowheads indicate RDF relations.
+			</p>
+			<div id="SHACL-UML.svg"></div>
```
Why is shacl12-core in this PR?

It's not right:

"Turtle serialization of the SHACL vocabulary" is repeated 3 times.  

"The content of the boxes under the class name" occurs twice.

`<div id="SHACL-UML.svg"></div>` occurs twice.

robert-david committed:
#665: Added links to definition of validation result (#686)

# Conflicts:
#	shacl12-sparql/index.html

robert-david committed:
Reset to gh-pages version

robert-david committed:
Removed redundant text

robert-david committed:
Merge branch 'gh-pages' into issue-163-PATH

robert-david commented:
The rebase went wrong. I hope i cleaned it up correctly now.

afs commented:
The PR changes  shacl12-core, shacl12-profiling and shacl12-sparql.
For example, the shacl12-core changes the shapes diagram and text related to it.

But the issue is only for shacl12-sparql.

It might be simpler to extract the changes related to `$PATH` (shacl12-sparql/index.html only) and make new PR.
